controller/array_action/array_mediator_ds8k.py

```diff
@@ -0,0 +1,412 @@
+                raise controller_errors.CredentialsError(self.service_address)
+            else:
+                raise ConnectionError()
+        except ClientException as e:
+            logger.error(
+                'Failed to connect to DS8K array {}, reason is {}'.format(
+                    self.service_address,
+                    e
+                    )
+                )
+            raise ConnectionError()
+
+    def disconnect(self):
+        pass
+
+    def get_system_info(self):
+        """Get the system result"""
+        system_info = self.client.get_system()[0]
+        return system_info
+
+    @property
+    def identifier(self):
+        if self._system_info is None:
+            self._system_info = self.get_system_info()
+        return self._system_info[SYSTEM_ID]
+
+    @property
+    def name(self):
+        if self._system_info is None:
+            self._system_info = self.get_system_info()
+        return self._system_info.get(SYSTEM_NAME, None) \
+            or self.identifier
+
+    @property
+    def version(self):
+        if self._system_info is None:
+            self._system_info = self.get_system_info()
+        return parse_version(self._system_info[SYSTEM_CODE_LEVEL])
+
+    @property
+    def wwnn(self):
+        if self._system_info is None:
+            self._system_info = self.get_system_info()
+        return self._system_info[SYSTEM_WWNN]
+
+    def _generate_volume_scsi_identifier(self, volume_id):
+        return '6{}000000000000{}'.format(self.wwnn[1:], volume_id)
+
+    def _generate_volume_response(self, res):
+        return Volume(
+            vol_size_bytes=int(res[VOLUME_LOGICAL_CAP]),
+            vol_id=self._generate_volume_scsi_identifier(res[VOLUME_ID]),
+            vol_name=res[VOLUME_NAME],
+            array_address=self.service_address,
+            pool_name=res[VOLUME_POOL_ID],
+            array_type=self.array_type
+        )
+
+    def create_volume(self, name, size_in_bytes, capabilities, pool_id):
+        logger.info(
+            "Creating volume with name: {}, size: {}, in pool: {}, "
+            "with capabilities: {}".format(
+                name, size_in_bytes, pool_id,
+                capabilities
+            )
+        )
+        try:
+            cli_kwargs = build_kwargs_from_capabilities(capabilities, pool_id,
+                                                        name, size_in_bytes)
+            logger.debug(
+                "Start to create volume with parameters: {}".format(cli_kwargs)
+            )
+            res = self.client.create_volumes(**cli_kwargs)[0]
+            if 'id' in res:
+                logger.info("finished creating volume {}".format(name))
+                return self._generate_volume_response(res)
+            elif RES_STATUS in res and \
+                    res[RES_STATUS].lower() not in RES_SUCCESS_STAUTS:
+                msg = 'Failed to create volume {} on array {}. {}'.format(
```

Comment:
`msg = 'Failed to create volume {} on array {}. {}'.format(name, self.identifier, res.get(RES_MSG, ''))` is a long line.
instead of breaking it between the arguments, you can make it into two lines:
```
err_msg_format = 'Failed to create volume {} on array {}. {}'
err_msg = err_msg_format.format(name, self.identifier, res.get(RES_MSG, ''))
```